javaslang/src/test/java/javaslang/collection/VectorPropertyTest.java

```diff
@@ -29,6 +29,20 @@ public void shouldCreateAndGet() {
             for (int j = 0; j < actual.size(); j++) {
                 assertThat(expected.get(j)).isEqualTo(actual.get(j));
             }
+
+            final Vector<Integer> actualInt = Vector.ofAll(LeafType.<int[]> asPrimitives(int.class, expected));
+            assert (i == 0) || (actualInt.trie.type.type() == int.class);
+            assertAreEqual(expected, actualInt);
+
+            final Seq<Character> expectedChar = expected.map(v -> (char) v.intValue());
```

Comment:
same for `char[]`, which doesn't have an optimized `LeafType` yet, just a generic one via reflection (currently only `int[]` arrays are optimized speed-wize, the rest are all delegated to `java.lang.reflect.Array`).
Each primitive can be optimized on demand, as done in `IntArray`.
@danieldietrich, would you like me to add other primitive speed optimizations in this commit also, or is `int[]` enough for now (the rest are also stored as primitives, but their usage is slower because of the reflective call)